src/zope/publisher/browser.py

```diff
@@ -167,37 +178,39 @@ def registerTypeConverter(field_type, converter, replace=False):
     type_converters[field_type] = converter
 
 
-isCGI_NAME = lambda key: key in {
+def isCGI_NAME(key): return key in {
     # These fields are placed in request.environ instead of request.form.
-    'SERVER_SOFTWARE' : 1,
-    'SERVER_NAME' : 1,
-    'GATEWAY_INTERFACE' : 1,
-    'SERVER_PROTOCOL' : 1,
-    'SERVER_PORT' : 1,
-    'REQUEST_METHOD' : 1,
-    'PATH_INFO' : 1,
-    'PATH_TRANSLATED' : 1,
-    'SCRIPT_NAME' : 1,
-    'QUERY_STRING' : 1,
-    'REMOTE_HOST' : 1,
-    'REMOTE_ADDR' : 1,
-    'AUTH_TYPE' : 1,
-    'REMOTE_USER' : 1,
-    'REMOTE_IDENT' : 1,
-    'CONTENT_TYPE' : 1,
-    'CONTENT_LENGTH' : 1,
+    'SERVER_SOFTWARE': 1,
+    'SERVER_NAME': 1,
+    'GATEWAY_INTERFACE': 1,
+    'SERVER_PROTOCOL': 1,
+    'SERVER_PORT': 1,
+    'REQUEST_METHOD': 1,
+    'PATH_INFO': 1,
+    'PATH_TRANSLATED': 1,
+    'SCRIPT_NAME': 1,
+    'QUERY_STRING': 1,
+    'REMOTE_HOST': 1,
+    'REMOTE_ADDR': 1,
+    'AUTH_TYPE': 1,
+    'REMOTE_USER': 1,
+    'REMOTE_IDENT': 1,
+    'CONTENT_TYPE': 1,
+    'CONTENT_LENGTH': 1,
     'SERVER_URL': 1,
```

Comment:
This is code from before Python had sets.  Let's just drop all the unused dict values and make this a set literal.

Reply:
Nice idea.